standard/structs.md

```diff
@@ -502,6 +504,50 @@ Automatically implemented properties ([§15.7.4](classes.md#1574-automatically-i
 
 > *Note*: This access restriction means that constructors in structs containing automatically implemented properties often need an explicit constructor initializer where they would not otherwise need one, to satisfy the requirement of all fields being definitely assigned before any function member is invoked or the constructor returns. *end note*
 
+A *property_declaration* ([§15.7.1](classes.md#1571-general)) for an instance property in a *struct_declaration* may contain the *property_modifier* `readonly`. However, a static property shall not contain that modifier.
+
+It is a compile-time error to attempt to modify the state of an instance struct variable via a readonly property declared in that struct.
```

Comment:
Agree with @jskeet and additionally we have some repetition here which I think should be factored out as it is for classes: under classes modifiers are generally described in subclauses under *§15.3 Class members*.

That same pattern can be repeated here with the current content of *§16.3 Struct members* becoming *§16.3.1 General* and adding *§16.3.2 The readonly modifier*.

In §16.3.2 the rules for readonly members can be given (not proposed text):

- They cannot mutate the state of the struct instance
- They may mutate the static state of the struct
- And the (subjectively) ugly one, a readonly member may call a mutating member (of the same instance only?) but any mutations made must not be observable after the call. This could in terms of the compiler strategy of “defensive copies”, see #1053, or not.

This would provide a single place for defining what a readonly member. The additional per-member differences could follow in §16.3 subclauses or remain as they are in §16.4 subclauses.